src/app/qgsmaptooldigitizefeature.cpp

```diff
@@ -138,21 +138,31 @@ void QgsMapToolDigitizeFeature::cadCanvasReleaseEvent( QgsMapMouseEvent *e )
       return;
     }
 
-    QgsPointXY savePoint; //point in layer coordinates
+    QgsPoint savePoint; //point in layer coordinates
+    bool isMatchPointZ = false;
     try
     {
       QgsPoint fetchPoint;
       int res;
       res = fetchLayerPoint( e->mapPointMatch(), fetchPoint );
+      if ( QgsWkbTypes::hasZ( fetchPoint.wkbType() ) )
+        isMatchPointZ = true;
+
       if ( res == 0 )
       {
-        savePoint = QgsPointXY( fetchPoint.x(), fetchPoint.y() );
+        if ( isMatchPointZ )
+          savePoint = QgsPoint( QgsWkbTypes::PointZ, fetchPoint.x(), fetchPoint.y(), fetchPoint.z() );
```

Comment:
Couldn't this simply be

`savePoint = fetchPoint;`?

Reply:
right